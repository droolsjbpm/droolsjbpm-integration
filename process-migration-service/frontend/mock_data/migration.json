{
    "id": 99,
    "definition": {
      "execution": {
        "type": "ASYNC",
        "callback_url": "This is mockup reponse",
        "scheduled_start_time": null
        },
      "plan_id": 2,
      "process_instance_ids": [
            1
        ]
    },
    "status": "CREATED",
    "created_at": "2019-01-12T01:04:52.035",
    "started_at": null,
    "finished_at": null,
    "cancelled_at": null,
    "error_message": null
}