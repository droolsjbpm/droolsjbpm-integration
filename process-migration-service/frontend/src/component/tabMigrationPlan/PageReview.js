import React, { Component } from "react";
import { Button } from "patternfly-react";

export default class PageReview extends Component {
  onDownloadClick = () => {
    var dataStr =
      "data:text/json;charset=utf-8," +
      encodeURIComponent(this.props.inputJsonStr);
    var dlAnchorElem = document.getElementById("downloadAnchorElem");
    dlAnchorElem.setAttribute("href", dataStr);
    dlAnchorElem.setAttribute("download", "migrationPlan.json");
    dlAnchorElem.click();
  };

  render() {
    return (
      <div>
        <div className="form-group">
          <Button onClick={this.onDownloadClick}>Export</Button>
          <pre>{this.props.inputJsonStr}</pre>
          <a id="downloadAnchorElem" style={{ display: "none" }} />
        </div>
      </div>
    );
  }
}
