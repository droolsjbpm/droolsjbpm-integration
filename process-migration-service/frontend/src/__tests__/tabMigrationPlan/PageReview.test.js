import renderer from "react-test-renderer";
import React from "react";
import PageReview from "../../component/tabMigrationPlan/PageReview";
import migration from "../../../mock_data/migration.json";

test("PageReview renders correctly using snapshot", () => {
  const tree = renderer
    .create(<PageReview inputJsonStr={JSON.stringify(migration)} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
