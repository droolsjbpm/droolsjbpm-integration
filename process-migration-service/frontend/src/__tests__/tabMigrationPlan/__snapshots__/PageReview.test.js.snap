// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageReview renders correctly using snapshot 1`] = `
<div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Export
    </button>
    <pre>
      {"id":99,"definition":{"execution":{"type":"ASYNC","callback_url":"This is mockup reponse","scheduled_start_time":null},"plan_id":2,"process_instance_ids":[1],"kieServerId":"sample-server"},"status":"CREATED","created_at":"2019-01-12T01:04:52.035","started_at":null,"finished_at":null,"cancelled_at":null,"error_message":null}
    </pre>
    <a
      id="downloadAnchorElem"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</div>
`;
